src/sbvr-api/control-flow.ts

```diff
@@ -0,0 +1,60 @@
+import * as _ from 'lodash'
+import * as Promise from 'bluebird'
+
+// TODO: We force the return type here to merge `Promise<U> | Promise<U[]>` into `Promise<U | U[]>,
+// which is equivalent for our uses but fixes issues where typing were being rejected
```

Comment:
Another point for requiring explicit function return types? :p

Reply:
Not really, `Promise<U> | Promise<U[]>` is the more accurate return type (it's obvious why when you think of an `Array<U> | Array<U[]>` case), it's just it was breaking types and in this particular case is safe to cast to the alternative form - I think I can switch to an even better typing though

Reply:
Ok, so the better typing was
```typescript
export const liftP = <T, U>(fn: (v: T) => U | Promise<U>) => {
	return ((a: T | T[]) => {
		if (_.isArray(a)) {
			// This must not be a settle as if any operation fails in a cs
			// we want to discard the whole
			return Promise.mapSeries(a, fn)
		} else {
			return Promise.resolve(a).then(fn)
		}
	}) as {
		(a: T[]): Promise<U[]>
		(a: T): Promise<U>
	}
}
```
which is valid typing and can differentiate the response correctly, and is generally the more accurate typing, but unfortunately won't work for our needs as it still ends up as `Promise<U> | Promise<U[]>`

Reply:
I didn't see your comment yesterday (I blame github's failed updating!) but I agree this case should always have used the return value to cast the return type.  I expect I just wasn't aware you could do it that way when I first wrote it